File: .local/bin/open.lua
#!/usr/bin/env lua

associations = {
  { filter = '.*.png$', command = "vimiv" },
  { filter = '.*.jpg$', command = "vimiv" },
  { filter = '.*.pdf$', command = "zathura" },
  { filter = '.*.mp3$', command = "mpv --force-window" },
  { filter = '.*.mp4$', command = "mpv --force-window" },
  { filter = '.*.svg$', command = "inkscape" },
  { filter = '.*.doc$', command = "libreoffice" }
}

for _, a in pairs(associations) do
  if string.match(arg[1], a.filter) then
    os.execute(a.command .. ' "' .. arg[1] .. '"')
    break
  end
end

File: .local/bin/open.ts
#!/usr/bin/env -S deno -q run --allow-run

const file = Deno.args[0]

const associations = [
  { filter: /.*\.(png|jpg)$/, command: ['vimiv'] },
  { filter: /.*\.pdf$/, command: ['zathura'] },
  { filter: /.*\.(mp3|mp4)$/, command: ['mpv', '--force-window'] },
  { filter: /.*\.svg$/, command: ['inkscape'] },
  { filter: /.*\.doc$/, command: ['libreoffice'] }
]

for (let { filter, command } of associations) {
  if (filter.test(file)) {
    await Deno.run({ cmd: [...command, file] }).status()
  }
}
